test/dhcp_libvirt/dhcp_libvirt_provider_test.rb

```diff
@@ -0,0 +1,86 @@
+require 'test_helper'
+require 'dhcp_libvirt/dhcp_libvirt'
+require 'dhcp_libvirt/dhcp_libvirt_main'
+require 'dhcp_common/dependency_injection/dependencies'
+
+class DhcpLibvirtProviderTest < Test::Unit::TestCase
+  def setup
+    fixture = <<XMLFIXTURE
+<network>
+  <name>default</name>
+  <uuid>25703051-f5d4-4a31-80b7-37bbbc4d19e1</uuid>
+  <forward mode='nat'>
+    <nat>
+      <port start='1024' end='65535'/>
+    </nat>
+  </forward>
+  <bridge name='virbr0' stp='on' delay='0'/>
+  <mac address='52:54:00:ed:a7:f7'/>
+  <ip address='192.168.122.1' netmask='255.255.255.0'>
+    <dhcp>
+      <range start='192.168.122.2' end='192.168.122.254'/>
+      <host mac="00:16:3e:77:e2:ed" name="foo-1.example.com" ip="192.168.122.10" />
+      <host mac="00:16:3e:77:e2:ee" name="foo-2.example.com" ip="192.168.122.11" />
+    </dhcp>
+  </ip>
+</network>
+XMLFIXTURE
+    @json_leases = [{
+        "ipaddr" => "192.168.122.22",
+        "mac" => "52:54:00:13:05:12",
+        "expirytime" => 1_455_723_598
+    }]
+    @libvirt_network = mock()
+    @libvirt_network.stubs(:dump_xml).returns(fixture)
+    @libvirt_network.stubs(:dhcp_leases).returns(@json_leases)
+    @subnet = Proxy::DHCP::Subnet.new("192.168.122.0", "255.255.255.0")
+    @service = Proxy::DHCP::SubnetService.new
+    @subnet_store = @service.subnets = Proxy::MemoryStore.new
+    @subject = ::Proxy::DHCP::Libvirt::Provider.new(:network => 'default', :libvirt_network => @libvirt_network, :name => "127.0.0.1")
+    @subject.initialize_for_testing(:service => @service)
+  end
+
+  def test_default_settings
```

Comment:
Probably worth adding a test verifying that the correct network class was used to initialize the provider?